web/satellite/src/components/accessGrants/CreateAccessModal.vue

```diff
@@ -0,0 +1,550 @@
+import BucketsSelection from '@/components/accessGrants/permissions/BucketsSelection.vue';
+import BucketNameBullet from "@/components/accessGrants/permissions/BucketNameBullet.vue";
+import CloseCrossIcon from '@/../static/images/common/closeCross.svg';
+import VCheckbox from '@/components/common/VCheckbox.vue';
+import TypesIcon from '@/../static/images/accessGrants/create-access_type.svg';
+import PermissionsIcon from '@/../static/images/accessGrants/create-access_permissions.svg';
+import NameIcon from '@/../static/images/accessGrants/create-access_name.svg';
+import BucketsIcon from '@/../static/images/accessGrants/create-access_buckets.svg';
+import DateIcon from '@/../static/images/accessGrants/create-access_date.svg';
+import NotesIcon from '@/../static/images/accessGrants/create-access_notes.svg';
+import Chevron from '@/../static/images/accessGrants/chevron.svg';
+
+import { ACCESS_GRANTS_ACTIONS } from '@/store/modules/accessGrants';
+import { BUCKET_ACTIONS } from "@/store/modules/buckets";
+import { APP_STATE_MUTATIONS } from "@/store/mutationConstants";
+
+import { RouteConfig } from '@/router';
+
+// @vue/component
+@Component({
+    components: {
+        VCheckbox,
+        VButton,
+        DurationSelection,
+        BucketsSelection,
+        BucketNameBullet,
+        CloseCrossIcon,
+        TypesIcon,
+        PermissionsIcon,
+        NameIcon,
+        BucketsIcon,
+        DateIcon,
+        NotesIcon,
+        Chevron,
+    },
+})
+export default class CreateAccessModal extends Vue {
+    @Prop({default: 'Default'})
+    private readonly label: string;
+    @Prop({default: 'Default'})
+    private readonly defaultType: string;
+    @Prop({default: () => { return; }})
+    private readonly onClose: () => void;
+
+    private checkedType = '';
+    private showAllPermissions = {show: false, position: "up"};
+    private permissionsList = ["read","write","list","delete"];
+    private accessName = '';
+    private checkedPermissions = {read: false, write: false, list: false, delete: false};
+    private selectedPermissions : string[] = [];
+    private allPermissionsClicked = false;
+    public areBucketNamesFetching = true;
+
+    /**
+     * Checks which type was selected on mount.
+     */
+    public async mounted(): Promise<void> {
+        this.checkedType = this.defaultType;
+        try {
+            await this.$store.dispatch(BUCKET_ACTIONS.FETCH_ALL_BUCKET_NAMES);
+            this.areBucketNamesFetching = false;
+        } catch (error) {
+            await this.$notify.error(`Unable to fetch all bucket names. ${error.message}`);
+        }
+    }
+
+    /**
+     * Closes modal.
+     */
+    public onCloseClick(): void {
+        this.$store.dispatch(ACCESS_GRANTS_ACTIONS.CLEAR_SELECTION);
+        this.onClose();
+    }
+
+    /**
+     * Toggles permissions list visibility.
+     */
+    public togglePermissions(): void {
+        if (this.showAllPermissions.show === false) {
```

Comment:
this.showAllPermissions.position = this.showAllPermissions.show ? "up" : "down";
this.showAllPermissions.show = !this.showAllPermissions.show;

Reply:
This was not accepted by the linter